### PR TITLE
Fix focus state of ComboBox

### LINKED ISSUE #8874: Combo Box doesn't loose focus when clicking away
### Environment Information 
 
- __Package version(s)__: 6.147.0
- __Browser and OS versions__: Chrome/Windows 10

### Please provide a reproduction of the bug in a codepen:
 
https://codepen.io/PatoBeltran/pen/yrrPdN

#### Actual behavior:
 
When having multiple combo box in the same page. If you open one, select an option and directly click to open another one, the focus styles on the first one stay and there's no way to remove them (as if the combo was focused at all times).

#### Expected behavior:

The combo box should loose focus as soon as the user clicks away from it.

#### Priorities and help requested:
 
Requested priority: (Blocking, High, Normal, Low) High

### PR DESCRIPTION
**Pull request by atneik**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #8874
- [x] Include a change request file using `$ npm run change`

#### Description of issue

`focus` state of the component is not in sync with the actual focus on the control. Due to lack of Blur handler on the button chevron down button, `focus` state still remains `true` while the actual focus is lost.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/8979)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx [hunk 1]
```diff
@@ -940,7 +942,9 @@ export class ComboBox extends BaseComponent<IComboBoxProps, IComboBoxState> {
    * OnBlur handler. Set the focused state to false
    * and submit any pending value
    */
-  private _onBlur = (event: React.FocusEvent<HTMLInputElement>): void => {
+  private _onBlur = (
+    event: React.FocusEvent<HTMLInputElement | Autofill | HTMLAnchorElement | HTMLButtonElement | HTMLDivElement | BaseButton | Button>
```
Comment (atneik) [inline packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: Is there a better way to fix type for the method?

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx [hunk 2]
```diff
@@ -410,6 +411,7 @@ export class ComboBox extends BaseComponent<IComboBoxProps, IComboBoxState> {
                 data-is-focusable={false}
                 tabIndex={-1}
                 onClick={this._onComboBoxClick}
+                onBlur={this._onBlur}
```
Comment (ecraig12345) [inline packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: Is there ever a situation where the icon can get blurred but some other part of the ComboBox still has focus?

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx (hunk 1 above)
Comment (KevinTCoughlin) [reply packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: You could widen the type to `React.FocusEvent<HTMLElement>`

Comment (ecraig12345) [reply packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: Trouble is that Autofill, BaseButton, and Button don't extend HTMLElement (I've been running into this same issue in other places)... In general I like having accurate typings, but in this case it seems rather obnoxious.

Comment (atneik) [reply packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: Yeah that's the problem, they all don't extend from `HTMLElement`. Either I could do what I have or rely on `any`.

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx (hunk 2 above)
Comment (atneik) [reply packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: When the `IconButton` is in focus, if you press Shift+Tab that will move the focus from the `IconButton` to `Autofill`. Even in this case `this.state.focused` on the component will still be consistent as Autofill's onFocus handler will take care of setting it back to true.

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx (hunk 1 above)
Comment (KevinTCoughlin) [reply packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: Sure, but the `HTMLElement` sub-classes could have be clumped under `HTMLElement` unioned with the non-`HTMLElement` types. Doesn't matter to me tho.

**Pull request merged**